RELEASE.md

```diff
@@ -1,5 +1,55 @@
 # Release Process
 
-Not doing releases yet.  
+The htm.core repository uses GitHub flow 
+https://guides.github.com/introduction/flow/ style development.
+Contributers to htm.core should be familiar with this Git strategy.
+
+We implement Semantic Versioning (i.e. MAJOR.MINOR.PATCH )
+See https://semver.org for a detailed definition but in gerneral
+it means Bug fixes not affecting the API increment the PATCH version, 
+backwards compatible API additions/changes (new features) increment the 
+MINOR version, and backwards incompatible API changes (breaking changes) 
+increment the MAJOR version. 
```

Comment:
I like this strategy, but we currently do not adhere to that. 

> ackwards incompatible API changes

depends what we consider the API in question. c++/py/NetworkAPI? 
If NetworkAPI, this is a sensible strategy, if c++, we still break it quite a while, so we'd skyrocket the major versions..

Reply:
> we still break it quite a while, so we'd skyrocket the major versions.

Perhaps we will not bump major versions every time...just on major changes or after a few of them accumulate. Eventually I expect this will settle down and breaking changes will not happen all that often. 

We can manually decide when we make a MAJOR or MINOR release.  Both can be made  manually by setting a tag in the master branch when we want them to occur.  No special coding or automatic setup to make that work.  Just a manual procedure.

The PATCH number  I want to increment automatically on each merge to master.   To do that I want to use the `mainline` option in GitVersion but I don't  know where that code should reside.  We don't use a build machine like Hudson where it would normally go. So I am thinking it would be in one of the CI's like travis....but it is not obvious as to how to do that.  Still reading.

Also, I want the new version to be placed in the 'VERSION' file so it can be picked up by the logic that deploys to PYPI when the master is build as a result of a merge.  I may have to write some sort of hook to make that happen.

Reply:
> Perhaps we will not bump major versions every time...just on major changes or after a few of them accumulate. Eventually I expect this will settle down

that sounds good. For now I'd say patch=no API breakage, minor = API incompatibility. Major = somethigh huge, manually decided. 

> I want to use the mainline option in GitVersion but I don't know where that code should reside

It'd be in the CI, for AppVeyor I've seen this already used. 

> the new version to be placed in the 'VERSION' file so it can be picked up by [...] PYPI when [...] may have to write some sort of hook to make that happen.

This is what NuPIC had, maybe the hook is still functionaning, or we can take the code from that repo.